src/middlewared/middlewared/plugins/kubernetes_linux/update.py

```diff
@@ -147,6 +176,37 @@ async def do_update(self, job, data):
 
         return await self.config()
 
+    @private
+    async def migrate_ix_applications_dataset(self, new_pool, old_pool):
+        snap_details = await self.middleware.call(
+            'zfs.snapshot.create', {
+                'dataset': applications_ds_name(old_pool),
+                'naming_schema': MIGRATION_NAMING_SCHEMA,
+                'recursive': True,
+            }
+        )
+
+        old_ds = applications_ds_name(old_pool)
+        new_ds = applications_ds_name(new_pool)
+        migrate_job = await self.middleware.call(
+            'replication.run_onetime', {
+                'direction': 'PUSH',
+                'transport': 'LOCAL',
+                'source_datasets': [old_ds],
+                'target_dataset': new_ds,
+                'recursive': True,
+                'also_include_naming_schema': [MIGRATION_NAMING_SCHEMA],
+                'retention_policy': 'NONE',
+                'replicate': True
+            }
+        )
+        await migrate_job.wait()
+        try:
+            if migrate_job.error:
+                raise CallError(f'Failed to migrate {old_ds} to {new_ds}: {migrate_job.error}')
+        finally:
+            await self.middleware.call('zfs.snapshot.delete', snap_details['id'], {'recursive': True})
```

Comment:
Shouldn't you cleanup all `MIGRATION_NAMING_SCHEMA` snapshots on both source and destination after the migration finishes?